OEL-1463: Update pager test assertions for first/last.

donquixote opened the pull request.

donquixote committed:
OEL-1463: Minimal fix for pager asserts, now that first/last was added.

donquixote committed:
OEL-1463: Simplify assertPagerLinks() parameters.

drishu commented on modules/oe_showcase_search/tests/src/ExistingSite/IntegrationTest.php:
```diff
@@ -225,32 +225,54 @@ protected function assertSearchResults(array $expected): void {
+    };
+
+    if ($active === $total - 1) {
+      $active_link = $pagination->find('css', 'li.active:last-child > a');
```
isn't it sufficient to call li.active to fetch the active link?

donquixote replied:
in this case the active and last link are identical. i want to verify that this is the case.

drishu replied:
you check the page index of the active link and then separately  for the last link, I dont see why you would care if its the same, but I wont insist, end result is the same so up 2 u

donquixote replied:
I agree now.
My earlier motivation was that I don't fully trust the pager, and wanted to be sure the last item is really the last link with the icon, not a bug where the last link is missing for some reason. This is why I wanted to check labels _and_ link destination.
But we cannot reliably check this anyway. So now I drop the label checks.

drishu commented on modules/oe_showcase_search/tests/src/ExistingSite/IntegrationTest.php:
```diff
@@ -225,32 +225,54 @@ protected function assertSearchResults(array $expected): void {
+
+    if ($active === $total - 1) {
+      $active_link = $pagination->find('css', 'li.active:last-child > a');
+      $last_link = NULL;
```
->find returns null if not found no ?
combined with above comment I think we can skip the if else and just grap the 2 things, then check for null or something

drishu commented on modules/oe_showcase_search/tests/src/ExistingSite/IntegrationTest.php:
```diff
@@ -225,32 +225,54 @@ protected function assertSearchResults(array $expected): void {
+    $pagination = $this->assertSession()
+      ->elementExists('css', 'ul.pagination');
+
+    $get_page_index = static function (NodeElement $link): string {
```
I agree that I would have also put it in a helper function, named or not, both are fine for me here, but do try to stick to certain scenarios when using them:
1. when there is no reusability within your project/you use it only in that specific place
2. as value of a callable parameter

at least this is my compass for it

donquixote committed:
OEL-1463: Replace the anonymous function.

donquixote committed:
OEL-1463: Document active link = last link.

donquixote committed:
OEL-1463: Explicitly cast to string when comparing.

This avoids weird cast to int in PHP 7.4 and lower, or for readers who think of PHP 7.4.

drishu commented on modules/oe_showcase_search/tests/src/ExistingSite/IntegrationTest.php:
```diff
@@ -225,32 +225,65 @@ protected function assertSearchResults(array $expected): void {
+   * @param \Behat\Mink\Element\NodeElement $pager_link
+   *   A link element that belongs to a pager.
+   */
+  protected function assertPagerLinkPageNumber(int $expected, NodeElement $pager_link): void {
```
I would have expected a string and typecasted when sending the value but not a biggy

donquixote replied:
I'd say semantically it is a number, and the `int` is more restrictive so gives a better idea what kind of value we expect.

donquixote replied:
Also I wanted to get the typecast out of the calling code, e.g. no `(string) ($total - 1)`.

donquixote committed:
OEL-1463: Simplify assertPager(), don't assert link text.

donquixote commented:
Pushed again, removed the commit in composer.json
